core/vm/evm.go

```diff
@@ -427,6 +436,10 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 	if evm.StateDB.GetNonce(address) != 0 || (contractHash != (common.Hash{}) && contractHash != emptyCodeHash) {
 		return nil, common.Address{}, 0, ErrContractAddressCollision
 	}
+	// Check whether the init code size has been exceeded.
+	if evm.Config.HasEip3860() && len(codeAndHash.code) > params.MaxInitCodeSize {
```

Comment:
This check being after the nonce bump does not look right to me. I think this should work as this check is performed at in the CREATE instruction (before switching context to the new account).